handler/src/test/java/io/netty5/handler/ssl/ApplicationProtocolNegotiationHandlerTest.java

```diff
@@ -200,17 +206,124 @@ public void channelReadComplete(ChannelHandlerContext ctx) {
         EmbeddedChannel channel = new EmbeddedChannel(new SslHandler(engine), new ChannelHandler() {
             @Override
             public void channelInboundEvent(ChannelHandlerContext ctx, Object evt) {
-                if (evt == SslHandshakeCompletionEvent.SUCCESS) {
+                if (evt instanceof SslHandshakeCompletionEvent && ((SslHandshakeCompletionEvent) evt).isSuccess()) {
                     ctx.fireChannelRead(someBytes);
                 }
                 ctx.fireChannelInboundEvent(evt);
             }
         }, alpnHandler);
-        channel.pipeline().fireChannelInboundEvent(SslHandshakeCompletionEvent.SUCCESS);
+        channel.pipeline().fireChannelInboundEvent(new SslHandshakeCompletionEvent(engine.getSession(), null));
         assertNull(channel.pipeline().context(alpnHandler));
         assertArrayEquals(someBytes, channelReadData.get());
         assertTrue(channelReadCompleteCalled.get());
         assertNull(channel.readInbound());
         assertTrue(channel.finishAndReleaseAll());
     }
+
+    private static final class TestSSLSession implements SSLSession {
```

Comment:
nit: Maybe we can use Mockito in this case?

Reply:
as we don't use any mocking here I was just doing it like this